command/agent/template/template.go

```diff
@@ -130,36 +142,49 @@ func (ts *Server) Run(ctx context.Context, incoming chan string, templates []*ct
 		select {
 		case <-ctx.Done():
 			ts.runner.Stop()
-			return
+			return nil
 
 		case token := <-incoming:
 			if token != *latestToken {
 				ts.logger.Info("template server received new token")
+
+				// If the runner was previously started and we intend to exit
+				// after auth, do not restart the runner if a new token is
+				// received.
+				if ts.exitAfterAuth && ts.runnerStarted.Load() {
+					ts.logger.Info("template server not restarting with new token with exit_after_auth set to true")
+					continue
+				}
+
 				ts.runner.Stop()
 				*latestToken = token
 				ctv := ctconfig.Config{
 					Vault: &ctconfig.VaultConfig{
 						Token: latestToken,
 					},
 				}
+
+				// If we're testing, limit retries to 3 attempts to avoid
+				// long test runs from exponential back-offs
+				if ts.testingLimitRetry != 0 {
+					ctv.Vault.Retry = &ctconfig.RetryConfig{Attempts: &ts.testingLimitRetry}
+				}
```

Comment:
I don't think you need the conditional:
https://github.com/hashicorp/consul-template/blob/7a4683109607e642f014f29898b1acc797efb890/config/retry.go#L46
https://github.com/hashicorp/consul-template/blob/master/config/vault.go#L196-L198

```suggestion
				ctv.Vault.Retry = &ctconfig.RetryConfig{Attempts: &ts.testingLimitRetry}
```

Reply:
The default value prior to this change is based off `DefaultRetryAttempts`, which is 12 retries and set by [Finalize](https://github.com/hashicorp/consul-template/blob/7a4683109607e642f014f29898b1acc797efb890/config/retry.go#L137). We don't want to change this to be unlimited by default.